modules/greenbidsAnalyticsAdapter.js

```diff
@@ -145,14 +191,30 @@ export const greenbidsAnalyticsAdapter = Object.assign(adapter({ANALYTICS_SERVER
       cachedAuction.timeoutBids.push(bid);
     });
   },
+  handleBillable(billableArgs) {
+    const vendor = billableArgs.vendor || 'unknown_vendor';
+    const billingId = billableArgs.billingId || 'unknown_billing_id';
+    /* Filter Greenbids Billable Events only */
+    if (vendor === 'greenbidsRtdProvider') {
+      this.billingId = billingId;
+    }
+  },
```

Comment:
IF you are only looking for greenbids vendor you prob do not need to set that local `vendor` var

```js
  handleBillable(billableArgs) {
    /* Filter Greenbids Billable Events only */
    if (billableArgs .vendor === 'greenbidsRtdProvider') {
      this.billingId = billableArgs.billingId || 'unknown_billing_id';
    }
  }
```


Also, this seems to be setting a module wide `billingId`

Which is emitted by Greenbids at `aucitonInit`

But it seems you do not send the message to your server until `auctionEnd`

Prebid supports running multiple auctions at once, so it is possible this data will not match up together.

If it matters to your server and stuff, you may want to have this function `handleBillable` save to a map of `auctionId => billingId`

Then in your `createBidMessage` you grab the `billingId` used for the auction your are sending.

Something like:
```js
  handleBillable(billableArgs) {
    /* Filter Greenbids Billable Events only */
    if (billableArgs .vendor === 'greenbidsRtdProvider') {
      this.billingIds[billableArgs.auctionId] = billableArgs.billingId || 'unknown_billing_id';
    }
  }
```

Reply:
Great feedback, it actually applied to more than `billingId` in the code, `greenbidsId` and `isSampled` were globally defined as well and not compatible with concurrent auctions. They have now been included in the `cachedAuctions` object which fixes this. Thanks for noticing !